### PR TITLE
Use InputFormatterException for another case

### PR DESCRIPTION
**Pull request by dougbu**
- #4917

### CODE REVIEW

File: src/Mvc/src/Microsoft.AspNetCore.Mvc.Formatters.Xml/XmlSerializerInputFormatter.cs [hunk 1]
```diff
@@ -159,6 +159,12 @@ public virtual InputFormatterExceptionPolicy ExceptionPolicy
             }
             // XmlSerializer wraps actual exceptions (like FormatException or XmlException) into an InvalidOperationException
             // https://github.com/dotnet/corefx/blob/master/src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs#L652
+            catch (InvalidOperationException exception) when (exception.InnerException != null &&
+                exception.InnerException.InnerException == null &&
+                string.Equals("Microsoft.GeneratedCode", exception.InnerException.Source, StringComparison.InvariantCulture))
+            {
```
Comment (dougbu) [inline src/Mvc/src/Microsoft.AspNetCore.Mvc.Formatters.Xml/XmlSerializerInputFormatter.cs]: FYI @Lxiamail @zhenlan

File: src/Mvc/test/Microsoft.AspNetCore.Mvc.FunctionalTests/SerializableErrorTests.cs [hunk 2]
```diff
@@ -89,25 +77,50 @@ public async Task PostedSerializableError_IsBound(string acceptHeader)
...
+                    string.Empty) +
+                ">" +
                 "<Id>2</Id><Name>foo</Name></Employee>";
             var expected = "<Error><Id>The field Id must be between 10 and 100.</Id>" +
                 "<Name>The field Name must be a string or array type with a minimum " +
                 "length of '15'.</Name></Error>";
             var request = new HttpRequestMessage(HttpMethod.Post, "http://localhost/SerializableError/CreateEmployee");
             request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse(acceptHeader));
-            request.Content = new StringContent(input, Encoding.UTF8, "application/xml-dcs");
+            request.Content = new StringContent(input, Encoding.UTF8, acceptHeader);
```
Comment (dougbu) [inline src/Mvc/test/Microsoft.AspNetCore.Mvc.FunctionalTests/SerializableErrorTests.cs]: This test previously exercised only the DCS input formatter. And, now that it tests both input formatters, need to deal with different namespace expectations of the underlying serializers. (Web site does not override the default DCS namespace for the `Employee` type.)

File: src/Mvc/src/Microsoft.AspNetCore.Mvc.Formatters.Xml/XmlSerializerInputFormatter.cs [hunk 3]
```diff
@@ -159,6 +159,12 @@ public virtual InputFormatterExceptionPolicy ExceptionPolicy
             }
             // XmlSerializer wraps actual exceptions (like FormatException or XmlException) into an InvalidOperationException
             // https://github.com/dotnet/corefx/blob/master/src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs#L652
+            catch (InvalidOperationException exception) when (exception.InnerException != null &&
```
Comment (rynowak) [inline src/Mvc/src/Microsoft.AspNetCore.Mvc.Formatters.Xml/XmlSerializerInputFormatter.cs]: Can you expand upon this in the comments? You did some valuable sleuthing here and I don't think it should be lost to the sands of time.

File: src/Mvc/test/Microsoft.AspNetCore.Mvc.FunctionalTests/SerializableErrorTests.cs (hunk 2 above)
Comment (rynowak) [reply src/Mvc/test/Microsoft.AspNetCore.Mvc.FunctionalTests/SerializableErrorTests.cs]: This is why theory tests make me sad.

**Review by rynowak — APPROVED** [review]

**Pull request merged**